docs/api-testing/use-cases/set-variable.md

```diff
@@ -0,0 +1,261 @@
+---
+id: set-variable
+title: Different ways to set a variable
+sidebar_label: Setting up Variables
+description: Different ways to set a variable
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+The **SET (Variable)** component allows you to create variables from a string to a payload. In this guide, you will learn about the various ways to create a variable.
+
+## What You'll Need
+
+- A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up)).
+- An existing API Testing Project. For details on how to create one, see [API Testing Quickstart](/api-testing/quickstart/).
+- Familiarity with the [API Testing Composer](/api-testing/composer/).
+
+## Mode: String
+
+This mode generates a String variable, which can be a static value or a variable taken from the response payload. If it is a static value you just have to write it and the engine will take it as is.
+
+Consider the following example:
+
+<img src={useBaseUrl('img/api-testing/set-string.png')} alt="Set static string" />
+
+```yaml
+- id: set
+  var: product
+  mode: string
+  value: t-shirt
+```
+
+In the above example the variable `product` will always have the value `t-shirt`.
+
+If you write it in ${...} brackets, you will make it dynamic and let the engine evaluate the value every time the test will be executed.
+
+Consider the following response payload that has been stored in the `payload` variable:
+
+```json
+{
+  "id": 1,
+  "name": "Baseball Cap",
+  "price": 29.99,
+  "category": "1",
+  "description": "This is product 1!",
+  "quantity": 5,
+  "imageURL": "http://image.com",
+  "color": [
+    "blue",
+    "yellow"
+  ],
+  "createdAt": "2021-11-28T21:58:43.461Z",
+  "updatedAt": "2021-11-28T21:58:43.461Z"
+}
+```
+
+If you write the following:
+
+<img src={useBaseUrl('img/api-testing/set-var-dynamic.png')} alt="Set dynamic value"/>
+
+```yaml
+- id: set
+  var: product
+  mode: string
+  value: ${payload.name}
+```
+
+The engine will evaluate the variable value every time the test will be executed. In the above scenario the variable `product` will contain the value `Baseball Cap` but if the response is the following:
+
+```json
+{
+  "id": 2,
+  "name": "Long Sleeve Shirt",
+  "price": 39.99,
+  "category": "1",
+  "description": "This is product 2!",
+  "quantity": 7,
+  "imageURL": "http://image.com",
+  "color": [
+    "blue",
+    "yellow",
+    "red"
+  ],
+  "createdAt": "2021-11-28T21:58:43.461Z",
+  "updatedAt": "2021-11-28T21:58:43.461Z"
+}
+```
+
+The value will be `Long Sleeve Shirt`, without changing your test.
+
+## Mode: Data
+
+Using this mode, the variable will be evaluated (like the [Expression](/api-testing/composer/expressions/) field), therefore the variable type can be everything. The variable type will depend on the object being evaluated.
```

Comment:
Not sure if this edit is correct, but I don't think "everything"  was the right word either. 
```suggestion
Using this mode, the variable will be evaluated (as ab [Expression](/api-testing/composer/expressions/) field), therefore the variable type can be any type. The variable type will depend on the object being evaluated.
```